src/utils.ts

```diff
@@ -281,3 +292,42 @@ export function smockableNode(node: ASTNode): boolean {
 
   return true;
 }
+
+/**
+ * Renders the abstract functions that are not implemented in the current contract
+ * @param contract The contract to render the abstract functions from
+ * @returns The content of the functions
+ */
+export async function renderAbstractUnimplementedFunctions(contract: ContractDefinition): Promise<string> {
+  let content = '';
+
+  if (contract.vLinearizedBaseContracts.length > 1) {
+    const variablesSelectors = contract.vStateVariables.map((variable) => variable.raw?.functionSelector);
+    const functionsSelectors = contract.vFunctions.map((func) => func.raw?.functionSelector);
+    const newFunctions = [];
+
+    for (const base of contract.vLinearizedBaseContracts) {
+      // Skip the first contract, which is the current contract
+      if (base.id === contract.id) continue;
+
+      for (const baseFunction of base.vFunctions) {
+        // Skip the functions that are already implemented in the current contract as variables
+        if (variablesSelectors.includes(baseFunction.raw?.functionSelector)) continue;
+
+        // Skip the functions that are already implemented in the current contract as functions
+        if (functionsSelectors.includes(baseFunction.raw?.functionSelector)) continue;
+
+        // If the function is already in the new functions array, skip it
+        if (newFunctions.some((func) => func.raw?.functionSelector === baseFunction.raw?.functionSelector)) continue;
+
+        newFunctions.push(baseFunction);
```

Comment:
Hmm we could define `newFunction` as a set and avoid the duplication check here.

Reply:
I tried it and failed, maybe because different base contract functions differ on some field

Maybe i made some mistake, I can try it again

Reply:
the `functionSelector` is the same but they have differents `id`